analytics_data_api/management/commands/tests/test_generate_stage_course_data.py

```diff
@@ -0,0 +1,108 @@
+import datetime
+
+from django.core.management import call_command
+from django.test import TestCase
+from freezegun import freeze_time
+
+from analytics_data_api.tests.test_utils import set_databases
+from analytics_data_api.v0 import models
+
+
+@set_databases
+class GenerateStageCourseDataTests(TestCase):
+
+    def test_run_with_no_preexisting_data(self):
+        """
+        Test that data is generated for the past four weeks
+        """
+        course_ids = [
+            'course-v1:edX+DemoX+Demo_Course', 'course-v1:edX+Insights+Stage', 'course-v1:edX+Insights+Fake+Data'
+        ]
+
+        call_command(
+            'generate_stage_course_data',
+            f"--database", 'analytics',
+        )
+
+        for model in [models.CourseEnrollmentDaily,
+                      models.CourseEnrollmentModeDaily,
+                      models.CourseEnrollmentByGender,
+                      models.CourseEnrollmentByEducation,
+                      models.CourseEnrollmentByCountry,
+                      models.CourseMetaSummaryEnrollment]:
+            for course_id in course_ids:
+                self.assertTrue(model.objects.filter(course_id=course_id).exists())
+
+                if model != models.CourseMetaSummaryEnrollment:
+                    # course meta summary generates a wider date range of data
+                    earliest_date = model.objects.filter(course_id=course_id).earliest('date').date
+                    self.assertEqual(earliest_date, datetime.date.today() - datetime.timedelta(weeks=4))
+
+        # check that Insights courses are in program table, while demo course is not
+        self.assertTrue(
+            models.CourseProgramMetadata.objects.filter(course_id='course-v1:edX+Insights+Stage').exists()
+        )
+        self.assertTrue(
+            models.CourseProgramMetadata.objects.filter(course_id='course-v1:edX+Insights+Fake+Data').exists()
+        )
+        self.assertFalse(
+            models.CourseProgramMetadata.objects.filter(course_id='course-v1:edX+DemoX+Demo_Course').exists()
+        )
+
+        self.assertFalse(models.CourseEnrollmentByBirthYear.objects.exists())
+
+    def test_run_with_preexisting_data(self):
+        """
+        Test that data is generated for the time between preexisting data and now,
+        e.g. if data was added a week ago, should only add a weeks worth of data.
+        No previous data should be deleted.
+        """
+        course_ids = [
+            'course-v1:edX+DemoX+Demo_Course', 'course-v1:edX+Insights+Stage', 'course-v1:edX+Insights+Fake+Data'
+        ]
```

Comment:
here again

I'm imagining updating them in the command and then updating these copies